volume/stream_in_out_nonlinux.go

```diff
@@ -41,7 +41,14 @@ func (streamer *tarGzipStreamer) Out(w io.Writer, src string, privileged bool) e
 }
 
 func (streamer *tarZstdStreamer) In(stream io.Reader, dest string, privileged bool) (bool, error) {
-	err := tarfs.Extract(zstd.NewReader(stream), dest)
+	zstdStreamReader := zstd.NewReader(stream)
+
+	err := tarfs.Extract(zstdStreamReader, dest)
+	if err != nil {
+		return true, err
+	}
+
+	err = zstdStreamReader.Close()
```

Comment:
Hey @kcmannem , I was wondering if there's a reason for the `Close` not being in a `defer` (or explicitly called in case of err above)?

I'm afraid we might still be missing some of the early allocations that the calls to `NewZstd...` might be causing and not be freed in case of errors.

wdyt?

thx!